### PR TITLE
Unique key refactor

### PR DESCRIPTION
**Pull request by dimagorelik**
New API is added to IdentifyerType createIdentifier(Entity entity)
UniqueKeyValue became protected class

### CODE REVIEW

File: main/src/main/java/com/kenshoo/pl/entity/UniqueKey.java [hunk 1]
```diff
@@ -54,6 +54,15 @@ public UniqueKey(Iterable<? extends EntityField<E, ?>> fields) {
...
     }
 
+    @Override
+    public Identifier<E> createIdentifier(Entity entity) {
+        Object[] values = new Object[fields.length];
+        for (int i = 0; i < fields.length; i++) {
+            values[i] = entity.get(fields[i]);
+        }
+        return new UniqueKeyValue<>(this, values);
+    }
```
Comment (effiban) [inline main/src/main/java/com/kenshoo/pl/entity/UniqueKey.java:64]: Can we use functional style (`Stream`) for this loop instead? Is there a performance issue?

Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/UniqueKey.java:64]: @effiban 
Do you prefer this code that was in the original method?
        final Object[] ids = Seq.of(key.getFields())
                .map(entity::get)
                .toArray();

Comment (effiban) [reply main/src/main/java/com/kenshoo/pl/entity/UniqueKey.java:64]: @dimagorelik - yes, I think that would be better

Comment (dimagorelik) [reply main/src/main/java/com/kenshoo/pl/entity/UniqueKey.java:64]: @effiban 
fixed

**Review by effiban — APPROVED** [review]

**Pull request merged**